### PR TITLE
Add GroupsCluster specs

### PR DESCRIPTION
**Pull request by Apollon77**

### CODE REVIEW

File: src/matter/cluster/GroupsCluster.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    EnumT, BitMapT, Bit
+} from "../../codec/TlvObjectCodec";
+import { Attribute, Cluster, Command, NoArgumentsT, NoResponseT } from "./Cluster";
+import { StatusCode } from "../interaction/InteractionMessages";
+import { GroupIdT } from "../common/GroupId";
+import { MatterApplicationClusterSpecificationV1_0 } from "../../Specifications";
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.3.7.1 */
+const AddGroupRequestT = ObjectT({
+    groupId: Field(0, GroupIdT), /* min: 1 */
```
Comment (mfucci) [inline src/matter/cluster/GroupsCluster.ts:22]: Should "min: 1" be part of GroupId ?
Or has GroupId = 0 a special meaning?

File: src/matter/cluster/GroupsCluster.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.3.7.3 */
+const GetGroupMembershipRequestT = ObjectT({
+    groupList: Field(0, ArrayT(GroupIdT)), /* groupId min: 1 */
+});
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.3.7.9 */
+const GetGroupMembershipResponseT = ObjectT({
+    /** contain the remaining capacity of the Group Table of the node. */
+    capacity: Field(0, UInt8T), /* nullable: true */
+    groupList: Field(1, ArrayT(GroupIdT)),
```
Comment (mfucci) [inline src/matter/cluster/GroupsCluster.ts:53]: Can GroupId be 0 here?

**Review by mfucci — APPROVED** [review]

File: src/matter/cluster/GroupsCluster.ts (hunk 2 above)
Comment (Apollon77) [reply src/matter/cluster/GroupsCluster.ts:53]: you are right, fixing

File: src/matter/cluster/GroupsCluster.ts (hunk 1 above)
Comment (Apollon77) [reply src/matter/cluster/GroupsCluster.ts:22]: bit I add min as Bound now

Comment (mfucci) [reply src/matter/cluster/GroupsCluster.ts:22]: Ok, let's keep it like that and fix it later when we rework Templates

**Review by mfucci — APPROVED** [review]

**Pull request merged**